### PR TITLE
Removing dynamics from EventProcessor

### PR DESCRIPTION
**Pull request by kapil-ms**

### CODE REVIEW

File: Common/Mapper/TypeMapper.cs [hunk 1]
```diff
@@ -41,6 +42,7 @@ private T typeConverter<T>(dynamic dynamicObj)
             if (typeof(T) == typeof(DeviceND))
             {
                 FixIsSimulatedDevice(dynamicObj);
+                DeviceSchemaHelper.FixDeviceSchema(dynamicObj);
```
Comment (kvish) [inline Common/Mapper/TypeMapper.cs]: use DeviceSchemaHelperND

Comment (kapil-ms) [inline Common/Mapper/TypeMapper.cs]: I am fixing the dynamic object here. ND version will not have this function at all

File: Common/DeviceSchema/CommandHistorySchemaHelper.cs [hunk 2]
```diff
@@ -176,6 +197,50 @@ private static int GetCommandHistoryItemIndex(dynamic commandHistory, string mes
 
             return result;
         }
+        private static int GetCommandHistoryItemIndexND(IList<CommandHistoryND> commandHistory, string messageId)
+        {
+            IEnumerable commands;
```
Comment (handeadiguzel) [inline Common/DeviceSchema/CommandHistorySchemaHelper.cs]: IEnumerable<Command> ?

File: Common/DeviceSchema/CommandHistorySchemaHelper.cs [hunk 3]
```diff
@@ -176,6 +197,50 @@ private static int GetCommandHistoryItemIndex(dynamic commandHistory, string mes
...
+            }
+
+            if (string.IsNullOrEmpty(messageId))
+            {
+                throw new ArgumentException(
+                    "messageId is a null reference or empty string.",
+                    "messageId");
+            }
+
+            if ((commands = commandHistory as IEnumerable) != null)
```
Comment (handeadiguzel) [inline Common/DeviceSchema/CommandHistorySchemaHelper.cs]: do you have to do this: as IEnumerable?

Comment (handeadiguzel) [inline Common/DeviceSchema/CommandHistorySchemaHelper.cs]: just iterate through: commandHistory  ?

File: Common/DeviceSchema/CommandHistorySchemaHelper.cs [hunk 4]
```diff
@@ -232,7 +297,16 @@ public static void UpdateCommandHistoryItem(dynamic device, dynamic command)
                 history[commandIndex] = command;
             }
         }
+        public static void UpdateCommandHistoryItemND(DeviceND device, CommandHistoryND command)
+        {
+            IList<CommandHistoryND> history = GetCommandHistoryND(device);
 
+            int commandIndex = GetCommandHistoryItemIndexND(history, (string)command.MessageId);
```
Comment (handeadiguzel) [inline Common/DeviceSchema/CommandHistorySchemaHelper.cs]: MessageId is already string

File: DeviceAdministration/Web/App_Start/Startup.Autofac.cs [hunk 5]
```diff
@@ -69,10 +68,7 @@ protected override void Load(ContainerBuilder builder)
             builder.RegisterType<AlertsLogic>().As<IAlertsLogic>();
 
             //Repositories
-            builder.RegisterType<IotHubRepository>().As<IIotHubRepository>();
             builder.RegisterType<IotHubRepositoryND>().As<IIotHubRepositoryND>();
-            builder.RegisterType<DeviceRegistryRepository>().As<IDeviceRegistryListRepository>();
-            builder.RegisterType<DeviceRegistryRepository>().As<IDeviceRegistryCrudRepository>();
```
Comment (handeadiguzel) [inline DeviceAdministration/Web/App_Start/Startup.Autofac.cs]: nice :)

**Pull request merged**